src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,512 @@
+        }
+    }
+}
+
+impl From<ArgumentsItem> for simple_extensions::ArgumentsItem {
+    fn from(value: ArgumentsItem) -> Self {
+        match value {
+            ArgumentsItem::EnumArgument(EnumArgument {
+                name,
+                description,
+                options,
+            }) => simple_extensions::ArgumentsItem::Variant0 {
+                name: Some(name),
+                description: Some(description),
+                options,
+            },
+
+            ArgumentsItem::ValueArgument(ValueArgument {
+                name,
+                description,
+                value,
+                constant,
+            }) => simple_extensions::ArgumentsItem::Variant1 {
+                name: Some(name),
+                description: Some(description),
+                value,
+                constant: Some(constant),
+            },
+
+            ArgumentsItem::TypeArgument(TypeArgument {
+                name,
+                description,
+                type_,
+            }) => simple_extensions::ArgumentsItem::Variant2 {
+                name: Some(name),
+                description: Some(description),
+                type_,
+            },
+        }
+    }
+}
+
+/// Parse errors for [simple_extensions::ArgumentsItem].
+#[derive(Debug, Error, PartialEq)]
+pub enum ArgumentsItemError {
+    /// Empty enumeration option.
+    #[error("empty enumeration option")]
+    EmptyEnumOption,
+
+    /// Empty optional field.
+    #[error("the optional field `{0}` is empty and should be removed")]
+    EmptyOptionalField(String),
+}
+
+/// Arguments that support a fixed set of declared values as constant arguments.
+#[derive(Clone, Debug, PartialEq)]
+pub struct EnumArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
+
+    /// List of valid string options for this argument.
+    options: Vec<String>,
+}
+
+impl EnumArgument {
+    fn parse(
+        name: Option<String>,
+        description: Option<String>,
+        options: Vec<String>,
+    ) -> Result<EnumArgument, ArgumentsItemError> {
+        if options.iter().any(String::is_empty) {
+            return Err(ArgumentsItemError::EmptyEnumOption);
+        }
+
+        Ok(EnumArgument {
+            name: ArgumentsItem::parse_name(name)?,
+            description: ArgumentsItem::parse_description(description)?,
+            options,
+        })
+    }
+}
+
+impl From<EnumArgument> for ArgumentsItem {
+    fn from(value: EnumArgument) -> Self {
+        ArgumentsItem::EnumArgument(value)
+    }
+}
+
+/// Arguments that refer to a data value.
+#[derive(Clone, Debug)]
+pub struct ValueArgument {
+    /// A human-readable name for this argument to help clarify use.
+    name: String,
+
+    /// Additional description for this argument.
+    description: String,
```

Comment:
These fields aren't required:
```suggestion
    /// A human-readable name for this argument to help clarify use.
    name: Option<String>,

    /// Additional description for this argument.
    description: Option<String>,
```

Reply:
For strings, I think we can use empty string if this field is absent to reduce the code complexity, for example, a user may need to check if the field is `None` or `Some("")`. What's your opinion here?

Reply:
For these fields I think we should keep the `Option`. I think it's the better way of capturing that these fields are not specified in the definition. Maybe they want to follow the docs to use `arg0`, `arg1` etc here and in that case I think using `Option` makes it clear that they should handle the `None` case (if they care about it). We could also choose to reject `Some("")` or map that `None` to make it easier for consumers?

Reply:
Sounds good. I'll keep the `Option`, and reject `Some("")`.